packages/snaps-simulator/src/features/simulation/middleware.ts

```diff
@@ -10,8 +14,48 @@ import type {
 export const methodHandlers = {
   // eslint-disable-next-line @typescript-eslint/naming-convention
   metamask_getProviderState: getProviderStateHandler,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  eth_requestAccounts: getAccountsHandler,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  eth_accounts: getAccountsHandler,
+};
+
+export type MiscMiddlewareHooks = {
+  getMnemonic: () => Promise<Uint8Array>;
 };
 
+/**
+ * A mock handler for account related methods that always returns the first address for the selected SRP.
+ *
+ * @param _request - Incoming JSON-RPC request, ignored for this specific handler.
+ * @param response - The outgoing JSON-RPC response, modified to return the result.
+ * @param _next - The json-rpc-engine middleware next handler.
+ * @param end - The json-rpc-engine middleware end handler.
+ * @param hooks - Any hooks required by this handler.
+ */
+async function getAccountsHandler(
+  _request: JsonRpcRequest<unknown>,
+  response: PendingJsonRpcResponse<unknown>,
+  _next: JsonRpcEngineNextCallback,
+  end: JsonRpcEngineEndCallback,
+  hooks: MiscMiddlewareHooks,
+) {
+  const { getMnemonic } = hooks;
+
+  const node = await BIP44CoinTypeNode.fromDerivationPath([
+    await getMnemonic(),
+    `bip32:44'`,
+    `bip32:60'`,
+  ]);
+
+  const deriveAddress = await getBIP44AddressKeyDeriver(node);
+
+  const { address } = await deriveAddress(0);
```

Comment:
You could also just derive the full derivation path using `BIP44Node`. `getBIP44AddressKeyDeriver ` is primarily useful when deriving many addresses.